tests/test_equivalence.py

```diff
@@ -20,13 +21,43 @@
 from fasm2bels.fasm2bels import main
 
 
-test_names = ["add32"]
+test_names = ["add32","alu"]
 
 
 def unpack_tar(tar_file):
     tar = tarfile.open(name=tar_file, mode="r:gz")
     tar.extractall(path=os.path.dirname(tar_file))
 
+def createGoldenfile(fileName):
+    path = os.environ['VIVADO_PATH'] #Note: Please use export VIVADO_PATH=/tools/Xilinx/Vivado/(VERSION OF VIVADO YOU'RE USING)/bin/vivado before running this.
+    file_exists = exists("equivalence_checking_data/"+fileName+"/top_bit.golden.v")
+    if(file_exists):
+        os.system("echo GoldenFile already exists.")
+        toTCL = "rm -r __pycache__"
+        os.system(toTCL)
+    else:
+        file_exists = exists(path)
+        if(file_exists):
+            updateTCL('FILENAME', fileName)
+            
+            os.system(path +  " -mode batch -source createGoldenFile.tcl")
+        
+            updateTCL(fileName, 'FILENAME')
+            
+            toTCL = "rm vivado* && rm -r *.dcp && rm -r *.edf && rm report_io.txt && rm -r .Xil && mv *_impl.v equivalence_checking_data/" + fileName + "/top_bit.golden.v"
+            os.system(toTCL)
+
+        else:
+            os.system("echo GoldenFile cannot be generated. Vivado must be installed in order for goldenfile to be created.")
```

Comment:
I think this can be avoided by just using arguments when calling Vivado. 

You can pass arguments to the tcl script by running:

```
vivado -mode batch -source <tcl_script_path>.tcl -tclargs <arg1> <arg2> <arg3> ...
```

In the tcl script you will then have:

```
set arg1 [lindex $argv 0]
set arg2 [lindex $argv 1]
set arg3 [lindex $argv 2]
...
```

In addition, you can also generate the file in a temporary directory/file and use it just for the purpose of the test, and the lifetime of the directory/file will only be limited to the execution of the script itself. You can check https://docs.python.org/3/library/tempfile.html for this purpose. The `create golden file` function can possibly return the path of this file or similar, so that it can get checked and than it will be automatically deleted.